File: Round_537_A.py
# code_report Solution
# Video Link: https://youtu.be/082mInmBC8Q
# Problem Link: https://codeforces.com/contest/1111/problem/A

def is_vowel(c):
    return c in ['a','e','i','o','u']

def superhero_transformed(a, b):
    if len(a) != len(b): return False
    res = True
    for c, d in zip(a, b):
        res = res and is_vowel(c) == is_vowel(d)
    return res
    
a, b = input().strip(), input().strip()
if superhero_transformed(a, b): print('Yes')
else:                           print('No')

File: Round_537_A.cpp
// code_report Solution
// Video Link: https://youtu.be/082mInmBC8Q
// Problem Link: https://codeforces.com/contest/1111/problem/A

#include <iostream>
#include <numeric>
#include <string>

using namespace std;

bool is_vowel(char c) {
   return c == 'a' || c == 'e' || c == 'i' || 
          c == 'o' || c == 'u';
}

bool superhero_transformed(string const& a, string const& b) {
   return a.size() == b.size() && 
      // would like to use C++17 transform_reduce here but not implemented yet
      inner_product(cbegin(a), cend(a), cbegin(b), true,
         std::logical_and{},
         [&](auto a, auto b) { return is_vowel(a) == is_vowel(b); });
}

int main() {
   string a, b;
   cin >> a >> b;
   cout << (superhero_transformed(a, b) ? "Yes\n" : "\No");
   return 0;
}
